Updates modeling docs for VLANs. (#5609)
nautobot/nautobot: nautobot/docs/user-guide/core-data-model/ipam/vlan.md

+/- 2.2.0
    - Replaced `location` ForeignKey field with `locations` ManyToManyField, allowing a VLAN to be assigned to multiple Locations.

## Modeling VLANs

With the update to Nautobot 2.2 that introduced the ability to have a single VLAN be associated to multiple locations, there needs to be some _recommended_ methods to model VLANs with the new functionality.

You should be looking to model VLANs in the same way that the layer 2 functionality is set up.

### Multiple Locations Using Same VLAN ID

> In many organizations the use of cookiecutter like VLAN architecture is leveraged. In that a location may have VLAN30 be for the same purpose (ie Wireless, User, Voice) at each location. In this scenario, you should have a VLAN30 for each location. Do **not** create a single VLAN and then associate it with multiple locations.

Reviewer: Can we provide some justification for this recommendation?

Reviewer: Maybe something like this?

```suggestion
In many organizations the use of cookiecutter like VLAN architecture is leveraged. In that a location may have VLAN30 be for the same purpose (ie Wireless, User, Voice) at each location. In this scenario, you should model a distinct VLAN30 in Nautobot for each Location. Creating a single VLAN and then associating it with multiple Locations would not accurately reflect the state of the layer 2 network.
```

### Multiple Locations, Stretched Layer 2

> Where you would want to take into account of the Many to Many capability where you can have a single VLAN that is associated with multiple locations is where the layer 2 environment is in fact stretched and devices within the VLAN are able to communicate via layer 2 mechanisms. Such designs include a campus network where you do have multiple locations but a shared VLAN across all of them for say wireless networks. Or in a data center fabric where the layer 2 is in fact stretched across the sites.

Reviewer: Perhaps?

```suggestion
You would want to have a single VLAN that is associated with multiple Locations when the layer 2 environment is in fact stretched and devices within the VLAN are able to communicate via layer 2 mechanisms. Such designs include a campus network where you do have multiple Locations but a shared VLAN across all of them, for example wireless networks. Or in a data center fabric where the layer 2 is in fact stretched across the sites.
```
